### PR TITLE
feat(repl): v4 support

### PR DESCRIPTION
**Pull request by gtm-nayan**

### CODE REVIEW

File: packages/repl/src/lib/workers/bundler/index.js [hunk 1]
```diff
@@ -434,7 +447,7 @@ async function bundle({ uid, files }) {
 				format: 'iife',
 				name: 'SvelteComponent',
 				exports: 'named',
-				sourcemap: true
+				sourcemap: 'inline'
```
Comment (benmccann) [inline packages/repl/src/lib/workers/bundler/index.js:450]: will this show up in the JS Output tab? I don't know that I'd want it to. Can we leave it as `true` or should we set it to `false`?

File: packages/repl/src/lib/workers/bundler/index.js [hunk 2]
```diff
@@ -32,20 +32,29 @@ self.addEventListener(
...
 
-				try {
-					importScripts(`${svelte_url}/compiler.js`);
-				} catch {
-					self.svelte = await import(/* @vite-ignore */ `${svelte_url}/compiler.mjs`);
+				const { version } = await fetch(`${svelte_url}/package.json`).then((r) => r.json());
+				console.log(`Using Svelte compiler version ${version}`);
+
+				if (version.startsWith('4')) {
+					// unpkg doesn't set the correct MIME type for .cjs files
```
Comment (benmccann) [inline packages/repl/src/lib/workers/bundler/index.js:42]:
```suggestion
					// unpkg doesn't set the correct MIME type for .cjs files
					// https://github.com/mjackson/unpkg/issues/355
```

File: packages/repl/src/lib/workers/compiler/index.js [hunk 3]
```diff
@@ -17,10 +17,19 @@ self.addEventListener(
...
 			case 'init':
-				try {
-					importScripts(`${event.data.svelte_url}/compiler.js`);
-				} catch (e) {
-					self.svelte = await import(/* @vite-ignore */ `${event.data.svelte_url}/compiler.mjs`);
+				const { svelte_url } = event.data;
+				const { version } = await fetch(`${svelte_url}/package.json`).then((r) => r.json());
+
+				if (version.startsWith('4')) {
+					// unpkg doesn't set the correct MIME type for .cjs files
```
Comment (benmccann) [inline packages/repl/src/lib/workers/compiler/index.js:24]:
```suggestion
					// unpkg doesn't set the correct MIME type for .cjs files
					// https://github.com/mjackson/unpkg/issues/355
```

File: packages/repl/src/lib/workers/patch_window.js [hunk 4]
```diff
@@ -0,0 +1 @@
+self.window = self; // hack for magic-sring and rollup inline sourcemaps
```
Comment (benmccann) [inline packages/repl/src/lib/workers/patch_window.js:1]:
```suggestion
self.window = self; // hack for magic-sring and rollup inline sourcemaps

```

File: packages/repl/src/lib/workers/bundler/index.js (hunk 1 above)
Comment (gtm-nayan) [reply packages/repl/src/lib/workers/bundler/index.js:450]: No. What's shown in the output tab is generated by the other worker, `compiler/index.js`.

**Review by PuruVJ — APPROVED** [review]

**Pull request merged**